### PR TITLE
Two hot path optimizations

### PR DESCRIPTION
**Pull request by jridgewell**
- Observable
  - `Array#forEach` is slow. Use for loops for hot code.
- Pass
  - No need to allocate a function every time. Create a bound `#pass`, use it.

Also a little style cleanup. I'll revert those if you want.

### CODE REVIEW

File: src/observable.js [hunk 1]
```diff
@@ -64,9 +62,11 @@ export class Observable {
    * @param {TYPE} event
    */
   fire(event) {
-    this.handlers_.forEach(handler => {
+    const handlers = this.handlers_;
+    for (let i = 0; i< handlers.length; i++) {
```
Comment (cramforce) [inline src/observable.js]: Nit: missing space

**Comment by erwinmombay** [comment]
@jridgewell good stuff 💯 . LGTM

**Comment by cramforce** [comment]
Love it.

LGTM

File: src/observable.js (hunk 1 above)
Comment (jridgewell) [inline src/observable.js]: Fixed.

**Pull request merged**